tests/test_service.py

```diff
@@ -616,6 +616,50 @@ def test_disable_wishlist(self):
                          "404 Not Found: Wishlist '{}' was not found."
                          .format(test_wishlist.id))
 
+    def test_delete_item_from_wishlist(self):
+        """ Delete a single item """
+        # create a wishlist with item
+        wishlist, items = self._create_items(1)
+        item = items[0]
+        resp = self.app.delete(
+            "/wishlists/{}/items/{}".format(wishlist.id, item.id),
+            content_type="application/json"
+        )
+        self.assertEqual(resp.status_code, status.HTTP_204_NO_CONTENT)
+        # make sure it's deleted
+        resp = self.app.get(
+            "/wishlists/{}/items/{}".format(wishlist.id, item.id),
+            content_type="application/json"
+        )
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_delete_item_if_wishlist_not_found(self):
+        """ Test delete_item if wishlist is not found """
+        wishlist = WishlistFactory()
+        item = ItemFactory()
+        resp = self.app.delete(
+            "/wishlists/{}/items/{}".format(wishlist.id, item.id),
+            content_type="application/json"
+        )
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+        data = resp.get_json()
+        self.assertEqual(data["message"],
+                         "404 Not Found: Wishlist '{}' was not found."
+                         .format(wishlist.id))
+
+    def test_delete_item_if_item_not_found(self):
+        """ Test delete_item if item is not found """
+        wishlist, items = self._create_items(1)
+        item = ItemFactory()
+        resp = self.app.delete(
+            "/wishlists/{}/items/{}".format(wishlist.id, item.id),
+            content_type="application/json"
+        )
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
```

Comment:
This test case should have failed as the response should be 204.

Reply:
Not sure about that. I feel like we should differentiate these two scenarios. Throw 404 if the item couldn't be found as no delete operation was performed. And respond with 204 if we managed to perform the delete operation.

Reply:
Delete always returns a 204, irrespective of a successful item found or not.

Reply:
Resolved with 204.